[Clarabel.cpp] Cherry-pick of https://github.com/oxfordcontrol/Clarabel.cpp/pull/52

We can drop this patch once we upgrade to a release that contains it.

From: Jeremy Nimmer <jeremy.nimmer@tri.global>
Date: Mon, 30 Dec 2024 17:48:20 -0800
Subject: [PATCH] Remove unncessary <iostream> include

Including iostream means introducing the static (global) constructors
and destructors for std::cin, std::cerr, and std::cout. That extra
init and fini code is undesirable when those streams are not actually
used.

--- include/cpp/CscMatrix.h
+++ include/cpp/CscMatrix.h
@@ -1,7 +1,6 @@
 #pragma once
 
 #include <cstdint>
-#include <iostream>
 #include <type_traits>
 
 namespace clarabel
